bpo-27777: cgi.FieldStorage can't parse simple body with Cont…

ar45 opened the pull request:
…ent-Length and no Content-Disposition

- If content length is present, read at most len bytes.
- When reading `read_lines_to_eof` use `read` instead of `readline`.
- Use `self.__write` even when content length is present and > 1000.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-27777](https://bugs.python.org/issue27777) -->
https://bugs.python.org/issue27777
<!-- /issue-number -->

ar45 committed:
Fixes issue 27777: cgi.FieldStorage can't parse simple body with Content-Length and no Content-Disposition

- If content length is present, read at most len bytes.
- When reading `read_lines_to_eof` use `read` instead of `readline`.
- Use `self.__write` even when content length is present and > 1000.

auvipy reviewed (changes requested):
can you check the failures?

ar45 committed:
Updated test_cgi.py

Renamed `test_fieldstorage_readline` to `test_fieldstorage_read`
and test `read` instead of `readline` since we use `read`.

ZackerySpytz commented on Misc/NEWS.d/next/Library/2019-02-05-16-57-27.bpo-27777.AKUxDR.rst:
```diff
@@ -0,0 +1,2 @@
+Fixes a bug in :mod:`cgi` when a request has a "Content-Length" header but
```
I think `FieldStorage` should be mentioned.

ar45 committed:
Update 2019-02-05-16-57-27.bpo-27777.AKUxDR.rst

Added mention of FieldStorage
